### PR TITLE
boost: Fix build

### PR DESCRIPTION
**Pull request by bshastry**

### CODE REVIEW

File: projects/boost/build.sh [hunk 1]
```diff
@@ -16,7 +16,9 @@
 ################################################################################
 
 # Build boost
-./bootstrap.sh && ./b2 headers
+CXXFLAGS="-stdlib=libc++ -pthread" LDFLAGS="-stdlib=libc++" \
```
Comment (inferno-chromium) [inline projects/boost/build.sh:19]: Did you mean to do CXXFLAGS="$CXXFLAGS ..." ?? I dont think we should remove existing CXXFLAGS.

Comment (Dor1s) [reply projects/boost/build.sh:19]: +1, otherwise we're losing coverage and sanitizers instrumentation

**Review by inferno-chromium — CHANGES_REQUESTED** [review]
Requesting changes, see comment above.

File: projects/boost/build.sh (hunk 1 above)
Comment (bshastry) [reply projects/boost/build.sh]: Done

**Comment by inferno-chromium** [comment]
Build failed, can you disable leaks during build process.

**Comment by bshastry** [comment]
> Build failed, can you disable leaks during build process.

Does it suffice to do `export ASAN_OPTIONS=detect_leaks=0` post build steps in `build.sh`?

**Comment by inferno-chromium** [comment]
> > Build failed, can you disable leaks during build process.
> 
> Does it suffice to do `export ASAN_OPTIONS=detect_leaks=0` post build steps in `build.sh`?

Still more issues, fuzzers crashes quickly

```
#2	INITED cov: 779 ft: 780 corp: 1/1b exec/s: 0 rss: 36Mb
libc++abi: terminating with uncaught exception of type boost::wrapexcept<boost::regex_error>: The repeat operator "*" cannot start a regular expression.  The error occurred while parsing the regular expression: '>>>HERE>>>*'.
AddressSanitizer:DEADLYSIGNAL
=================================================================
==105==ERROR: AddressSanitizer: ABRT on unknown address 0x000000000069 (pc 0x7fe2859e8438 bp 0x7ffd7c6b3050 sp 0x7ffd7c6b2988 T0)
SCARINESS: 10 (signal)
    #0 0x7fe2859e8438 in raise (/lib/x86_64-linux-gnu/libc.so.6+0x35438)
    #1 0x7fe2859ea039 in abort (/lib/x86_64-linux-gnu/libc.so.6+0x37039)
    #2 0x71b755 in abort_message (/tmp/not-out/boost_regex_fuzzer+0x71b755)
    #3 0x7204fe in demangling_terminate_handler() (/tmp/not-out/boost_regex_fuzzer+0x7204fe)
    #4 0x71b1c2 in std::__terminate(void (*)()) (/tmp/not-out/boost_regex_fuzzer+0x71b1c2)
    #5 0x71cfa5 in __cxxabiv1::failed_throw(__cxxabiv1::__cxa_exception*) (/tmp/not-out/boost_regex_fuzzer+0x71cfa5)
    #6 0x71cf3f in __cxa_throw (/tmp/not-out/boost_regex_fuzzer+0x71cf3f)
    #7 0x680cbf in void boost::throw_exception<boost::regex_error>(boost::regex_error const&) /src/boost/./boost/throw_exception.hpp:161:5
    #8 0x680a25 in boost::regex_error::raise() const /src/boost/libs/regex/src/regex.cpp:82:4
    #9 0x5a6bd6 in boost::re_detail_107600::basic_regex_parser<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::fail(boost::regex_constants::error_type, long, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, long) /src/boost/./boost/regex/v4/basic_regex_parser.hpp:246:9
    #10 0x5766fa in boost::re_detail_107600::basic_regex_parser<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::fail(boost::regex_constants::error_type, long, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) /src/boost/./boost/regex/v4/basic_regex_parser.hpp:72:7
    #11 0x575b59 in boost::re_detail_107600::basic_regex_parser<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::parse_extended() /src/boost/./boost/regex/v4/basic_regex_parser.hpp:361:10
    #12 0x5769e4 in boost::re_detail_107600::basic_regex_parser<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::parse_all() /src/boost/./boost/regex/v4/basic_regex_parser.hpp:264:16
    #13 0x5710be in boost::re_detail_107600::basic_regex_parser<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::parse(char const*, char const*, unsigned int) /src/boost/./boost/regex/v4/basic_regex_parser.hpp:182:18
    #14 0x55b768 in boost::re_detail_107600::basic_regex_implementation<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::assign(char const*, char const*, unsigned int) /src/boost/./boost/regex/v4/basic_regex.hpp:233:14
    #15 0x55b302 in boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::do_assign(char const*, char const*, unsigned int) /src/boost/./boost/regex/v4/basic_regex.hpp:677:10
    #16 0x55b0a0 in boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::assign(char const*, char const*, unsigned int) /src/boost/./boost/regex/v4/basic_regex.hpp:398:14
    #17 0x55b00e in boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >& boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::assign<std::__1::char_traits<char>, std::__1::allocator<char> >(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int) /src/boost/./boost/regex/v4/basic_regex.hpp:436:14
    #18 0x558600 in boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::basic_regex<std::__1::char_traits<char>, std::__1::allocator<char> >(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int) /src/boost/./boost/regex/v4/basic_regex.hpp:411:7
    #19 0x5580d1 in LLVMFuzzerTestOneInput /src/boost/../boost_regex_fuzzer.cc:28:16
    #20 0x45b7e1 in fuzzer::Fuzzer::ExecuteCallback(unsigned char const*, unsigned long) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerLoop.cpp:599:15
    #21 0x45acea in fuzzer::Fuzzer::RunOne(unsigned char const*, unsigned long, bool, fuzzer::InputInfo*, bool, bool*) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerLoop.cpp:505:3
    #22 0x45c9f7 in fuzzer::Fuzzer::MutateAndTestOne() /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerLoop.cpp:745:19
    #23 0x45d585 in fuzzer::Fuzzer::Loop(std::__Fuzzer::vector<fuzzer::SizedFile, fuzzer::fuzzer_allocator<fuzzer::SizedFile> >&) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerLoop.cpp:883:5
    #24 0x44b86c in fuzzer::FuzzerDriver(int*, char***, int (*)(unsigned char const*, unsigned long)) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerDriver.cpp:906:6
    #25 0x4757e2 in main /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerMain.cpp:20:10
    #26 0x7fe2859d383f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x2083f)
    #27 0x41fe68 in _start (/tmp/not-out/boost_regex_fuzzer+0x41fe68)

DEDUP_TOKEN: raise--abort--abort_message
AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: ABRT (/lib/x86_64-linux-gnu/libc.so.6+0x35438) in raise
==105==ABORTING
MS: 2 ChangeBit-InsertByte-; base unit: adc83b19e793491b1c6ea0fd8b46cd9f32e592fc
0x2a,0x7e,
*~
artifact_prefix='./'; Test unit written to ./crash-693031f7d7c0b2d966be722223216a6062d5e998
Base64: Kn4=
```

**Pull request closed without merging**